src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -180,8 +180,8 @@ private static void HandleBinaryOperator(in OperationAnalysisContext context, IN
                     break;
             }
 
-            static bool IsNotInstanceInvocationOrNotOnSymbol(IInvocationOperation invocationOperation, INamedTypeSymbol symbolType)
-                => invocationOperation.Instance is null || IsSymbolType(invocationOperation.Instance, symbolType);
+            static bool IsSymbolInstance(IInvocationOperation invocationOperation, INamedTypeSymbol symbolType)
+                => invocationOperation.Instance is not null && IsSymbolType(invocationOperation.Instance, symbolType);
```

Comment:
💭 I believe the old method is the one we want for `Equals`, while the new one is the one we want for `GetHashCode`.

Reply:
> I believe the old method is the one we want for Equals

Won't old method will always return true for the static `object.Equals(obj1, obj2)`, which isn't desired?